### ISSUE TITLE
input file parameter

### ISSUE THREAD

**Issue by tolot27** [wdecoster/nanofilt#22]
It would be nice to have an input file parameter like `-i` or `--fastq` (compatible to nanopack) which accepts (probably compressed) file names. IMHO, this would better fit into pipelines like snakemake, and avoids an extra gzip process.

**Comment by wdecoster** [comment]
That can be done, but essentially you would just move the gzip inside NanoFilt, so the amount of 'work' remains the same. The current solution is as flexible as possible. In snakemake your shell command can contain pipes, so you can still use `gunzip -c {input} | NanoFilt | gzip > {output} `

If you would like to contribute to my scripts you are absolutely welcome to do so ;-)

**Comment by tolot27** [comment]
>If you would like to contribute to my scripts you are absolutely welcome to do so ;-)

I'll definitively do that - after I've finished my thesis. :-)

**Comment by wdecoster** [comment]
Then I'll leave this issue open for you :) It doesn't require big changes.

**Comment by peflanag** [comment]
Hi, I have no background in terminal command, I'm a microbiologist. But i recently started working with ONT. I just want to run Nanofilt on my porechoped fastq file. Its not zipped. How do I do it? With other command lines I've used I've typed something like this

NanoFilt -i {file to run in this case the porechop fastq file} -q 9 -l 1000 -o {folder to put output fastq in}

This probably sounds simple but I'm not command line savvy and I don't understand how to use Nanofilt cause it asks for gunzip. Can anyone help please?


Cheers,

Peter

**Comment by wdecoster** [comment]
Thanks for your question. See also https://github.com/wdecoster/nanofilt/issues/16

TLDR you can just replace gunzip by `cat` if your file is not compressed. But there is no real reason _not_ to compress your file, your hard drive will thank you :-)

**Comment by peflanag** [comment]
Oh ok thanks! I didn't realise I could just do the command gunzip [file] and it would zip it! Jusrt had a google there! So if i want to carry forward the zipped NanoFilt file into canu, which I know is a different command line, do i have to do gunzip -c [file name] | canu [fill in rest of script] ?


Cheers!

**Comment by wdecoster** [comment]
Let me take a step backward here:

- gzip is used to compress and create a .gz file
- gunzip is used to decompress

- gunzip -c also decompresses, but writes the output to stdout, directly "into" NanoFilt as input. For that you use the unix pipe symbol `|`.

- I don't think your canu command will work, you probably can use canu directly with the compressed file. Different tools use different approaches on how to handle input and output data, the documentation should make that clear.

**Comment by peflanag** [comment]
Cool I'll have a look and see! Cheers

**Comment by wdecoster** [comment]
Done in v2.3.0, see also https://github.com/wdecoster/nanofilt/issues/29
